src/blas/KokkosBlas2_teamvector_gemv.hpp

```diff
@@ -0,0 +1,89 @@
+/*
+//@HEADER
+// ************************************************************************
+//
+//                        Kokkos v. 3.0
+//       Copyright (2020) National Technology & Engineering
+//               Solutions of Sandia, LLC (NTESS).
+//
+// Under the terms of Contract DE-NA0003525 with NTESS,
+// the U.S. Government retains certain rights in this software.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are
+// met:
+//
+// 1. Redistributions of source code must retain the above copyright
+// notice, this list of conditions and the following disclaimer.
+//
+// 2. Redistributions in binary form must reproduce the above copyright
+// notice, this list of conditions and the following disclaimer in the
+// documentation and/or other materials provided with the distribution.
+//
+// 3. Neither the name of the Corporation nor the names of the
+// contributors may be used to endorse or promote products derived from
+// this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY NTESS "AS IS" AND ANY
+// EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+// PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL NTESS OR THE
+// CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+// PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+// LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+// NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+// SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// Questions? Contact Siva Rajamanickam (srajama@sandia.gov)
+//
+// ************************************************************************
+//@HEADER
+*/
+
+#ifndef KOKKOSBLAS2_TEAMVECTOR_GEMV_HPP_
+#define KOKKOSBLAS2_TEAMVECTOR_GEMV_HPP_
+
+#include "KokkosBatched_Gemv_Decl.hpp"
+#include <KokkosKernels_Error.hpp>
+
+namespace KokkosBlas {
+namespace Experimental {
+
+template <class TeamType, class MatrixType, class XVector, class YVector,
+          class ScalarType = typename MatrixType::non_const_value_type>
+void KOKKOS_INLINE_FUNCTION team_vector_gemv(
+    const TeamType& team, const char trans,
+    const typename MatrixType::non_const_value_type& alpha, const MatrixType& A,
+    const XVector& x, const typename YVector::non_const_value_type& beta,
+    const YVector& y) {
+  static_assert(
+      std::is_same<ScalarType, typename XVector::non_const_value_type>::value &&
+          std::is_same<ScalarType,
+                       typename YVector::non_const_value_type>::value,
+      "TeamVector GEMV requires A,x and y to have same scalar type");
+
+  const auto run = [&](auto mode) {
```

Comment:
similar comment as in #1433

Reply:
Thank you for pointing this out - fixed on https://github.com/kokkos/kokkos-kernels/pull/1435/commits/4bb49548e7c34ea9f53920668761b76ca389d804.